core/spec/models/spree/order_spec.rb

```diff
@@ -919,15 +919,23 @@ def compute(computable)
       expect { shipment.reload }.not_to raise_error
     end
 
-    it "raises an error if any shipments are shipped" do
-      shipment = create(:shipment, order: subject, state: "shipped")
+    it "raises an error if some shipments are shipped" do
+      shipped_shipment = create(:shipment, order: subject, state: "shipped")
+      pending_shipment = create(:shipment, order: subject, state: "pending")
       expect {
         expect {
           subject.create_proposed_shipments
         }.to raise_error(Spree::Order::CannotRebuildShipments)
       }.not_to change { subject.reload.shipments }
 
-      expect { shipment.reload }.not_to raise_error
+      expect { shipped_shipment.reload }.not_to raise_error
+    end
+
+    it "returns the shipments when all shipments are shipped" do
```

Comment:
This test tests the behaviour change, but doesn't expose why this change was necessary. Can we get a spec showing the problem with an unreturned exchange?